extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -63,11 +59,9 @@ MongoClientBuildItem build(BuildProducer<FeatureBuildItem> feature, MongoClientR
 
         sslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(FeatureBuildItem.MONGODB_CLIENT));
 
-        RuntimeValue<MongoClient> client = recorder.configureTheClient(config, beanContainer.getValue(),
+        recorder.configureTheClient(config, beanContainer.getValue(),
                 launchMode.getLaunchMode(), shutdown,
                 codecs.getCodecProviderClassNames());
-        RuntimeValue<ReactiveMongoClient> reactiveClient = recorder.configureTheReactiveClient();
-        return new MongoClientBuildItem(client, reactiveClient);
```

Comment:
That allows another extension to get the client. Why removing this feature? 
Not sure if Panache Mongo requires it But I had something using this (for demo purpose, so no problem to break it).

Reply:
Mongo Panache is not using this (at least as far as I could tell and when I ran the tests locally nothing failed).

I removed it because nothing was using it and the implementation threw me off. If you feel it's valid to keep it around for future use, then I have absolutely no problem closing this PR